pkg/v2/tkr/webhook/cluster/tkr-resolver/cluster/cluster.go

```diff
@@ -261,43 +268,98 @@ func (e *errUnresolved) Error() string {
 	return sb.String()
 }
 
-// setMetadata sets cluster TKR resolution metadata based on result.
+// setTKRData sets cluster TKR resolution metadata based on result.
 // It also ensures OSImage labels and os-image-ref annotation are set on controlPlane and machineDeployment
 // if TKR resolution took place.
-func setMetadata(result data.Result, cluster *clusterv1.Cluster) {
+func (cw *Webhook) setTKRData(result data.Result, cluster *clusterv1.Cluster) error {
 	if result.ControlPlane != nil {
+		tkrData, err := ensureTKRData(cluster)
+		if err != nil {
+			return err
+		}
+
 		getMap(&cluster.Labels)[runv1.LabelTKR] = result.ControlPlane.TKRName
```

Comment:
Question: Is it OK to update the label `LabelTKR`, doesn't it cause roll out?

Reply:
Cluster labels don't cause the rollout.